File: 프로그래머스/연습문제/Lv1/덧칠하기/덧칠하기.py
def solution(n, m, section):
    painted = [True for i in range(n)]
    for s in section: painted[s-1] = False

    cnt = 0
    for i in range(len(painted)):
        if not painted[i]:
            cnt += 1
            for j in range(m):
                if j+i >= len(painted): break
                painted[i+j] = True

    return cnt

File: 프로그래머스/연습문제/Lv1/덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    vector<bool> painted(n, true);
    for(int s : section) painted[s-1] = false;
    
    int cnt = 0;
    for(int i=0; i<painted.size(); i++){
        if(!painted[i]){
            cnt++;
            for(int j=0; j<m && j+i<painted.size(); j++)
                painted[i+j] = true;
        }
    }
    
    return cnt;
}
